### DISCUSSION TITLE
Supported NFC Devices

### DISCUSSION THREAD

**Discussion by digital-idiot** [solokeys/solo2#165] [General]
There seem to be a lot of confusion regarding NFC supported device by Solokey V2. I couldn't find any official list of supported NFC devices or general conditions for NFC to work properly.

I tested my two solo keys with my Google Pixel 5 (Android 13), none of them work through NFC or USB. To be clear the devices function as intended when used in laptop via USB but not on my phone. After browsing couple of threads on Reddit with similar complaints I learned about the idea to try NFC while the key is powered through USB. This hack worked as the mobile was able to detect both keys. However, obviously this is not the ideal way to use these keys and defeats the purpose of having NFC in the first place.

So my questions are:
1. Is it a hardware issue, a firmware issue or a comparability issue.
2. Is it fixable either via solo firmware update or future android patch?
3. I saw people reported same problem with Solokey V1. Assuming this is a hardware issue / insufficient power issue, how come this hasn't been addressed design wise at the time of inception of Solokey V2?
4. Could it be that my keys have some manufacturing defect and NFC issue is a manifestation of that? In that case I need replacement.

I'll be grateful if someone from solo or someone more knowledgeable can chime in and help with this.

**Comment by msiqx9** [comment]
This seems to be an issue with Android 13 with keys generally.  A few weeks ago I set up a new OnePlus phone with Android 13.1 and I had problems using a Yubico USB-C/NFC key so I had to set up my Google account with a recovery procedure without the key.

**Reply by digital-idiot** [reply]
I understand, if separately powered via USB the keys are being detected by the phone although the authentication fails. This is likely and Android bug. But what about issue of the keys not being detected at all without separately powered through USB?

I also encountered another issue, when the keys are connected upside down the touch authentication doesn't get registered. I can see that the LED turns blue on touch but the authentication does not go forward. Tested this on Linux 6.4 with both Firefox and Google Chrome.

**Comment by msiqx9** [comment]
> I also encountered another issue, when the keys are connected upside down the touch authentication doesn't get registered. I can see that the LED turns blue on touch but the authentication does not go forward. Tested this on Linux 6.4 with both Firefox and Google Chrome.

I have never encountered this problem.  I tested my Solo v2 key upside-down with Ubuntu 23-04 and it still worked.  I would suggest its a particular issue with USB socket in your PC.

**Comment by alexjmoore** [comment]
Is there any update on this topic?  As per the original author, Solo2 has never worked on Android for me, very frustrating.

**Comment by digital-idiot** [comment]
After several firmware update I can still confirm that NFC on Solo2 does not work as intended (i. e. without external power through USB).

**Comment by JuliusFreudenberger** [comment]
I tested my Solo 2 after I received my package.
I can report that both USB-C and NFC work with Android 13 (GrapheneOS installed on Google Pixel 6) and with Arch Linux.
No problems whatsoever until now.

**Comment by digital-idiot** [comment]
> I tested my Solo 2 after I received my package. I can report that both USB-C and NFC work with Android 13 (GrapheneOS installed on Google Pixel 6) and with Arch Linux. No problems whatsoever until now.

@JuliusFreudenberger  Is your key USB-A or USB-C edition? Both of my keys are USB-A. I am asking this because I want find out whether it is a manufacturing issue.

Clearly my phone has no NFC issue as it can read the key through NFC while the key is separately powered through USB. I also regularly pay using NFC and read various different RFIDs on the regular basis without any problem.

 If indeed this is a manufacturing defect then Solo devs must address this.

**Reply by JuliusFreudenberger** [reply]
As I wrote, it is the USB-C version.
If it is a manufacturing defect, the devs cannot do anything about it. You will have to get a replacement key, as it is not a software issue but a hardware issue.

**Reply by digital-idiot** [reply]
Thanks for clarifying. I did email them immediately asking for clarification. I never received any response from them. Now I emailed them again requesting replacement. It is a bit unfortunate to receive this kind of unreliable products and lackluster support.

![image](https://github.com/solokeys/solo2/assets/10576475/0f65cc1d-fafa-42e7-b507-e6433cf9000e)


![image](https://github.com/solokeys/solo2/assets/10576475/ffb99d70-da9d-403b-863b-d204018bf99b)

**Reply by digital-idiot** [reply]
@JuliusFreudenberger Have you set PIN for your keys? Apparently:

> There were comments in the indiegogo that to make NFC work one would need to disable pin and it's android limitation, not solo's. Unfortunately I don't know how to disable pin without resetting the key.

https://github.com/solokeys/solo2/discussions/174#discussioncomment-7091582

**Comment by schildbach** [comment]
It does not work for me either, Solo2 USB-C against Pixel 4a (5G) – it detects my key but then "authentication failed". On the USB-C port of the phone the key starts blinking erratically blue but the buttons don't react. No problems on an USB-C port of my Linux.-powered Dell laptop though. I hope the firmware will be improved at some day…

**Reply by axel37** [reply]
Similar behaviour on my Galaxy S22 :
- NFC : When it's detected, I get an error in the app trying to authenticate
- USBC : I'm stuck at the "connect key" screen stating "if your key has a gold pad, tap it now", with the led blinking randomly

**Comment by alexjmoore** [comment]
NFC has never worked with either my standard USB-C key or my limited edition one, if it is a fault, I'm getting the impression it must be a pretty widespread issue.  At this point I'm not confident the solo2 keys can be relied on at all.

**Reply by digital-idiot** [reply]
> if it is a fault, I'm getting the impression it must be a pretty widespread issue. At this point I'm not confident the solo2 keys can be relied on at all.

Exactly. It is infuriating that Devs are completely silent on this. Neither they are responding here nor via official email. Initially I gave them the benefit of doubt but it cannot be a coincidence that so many people are having the same issue.

**Comment by BarsMonster** [comment]
Can anyone with issues try NFC without pin?

**Reply by digital-idiot** [reply]
I did. I completely reset one of my keys through _**chrome://settings/securityKeys**_, then I verified it whether it has actually been reset by trying to login to one of the website where it was previously registered. I was notified that the key is not registered, which confirms that the key is actually reset. The PIN which was set previously is also gone. After all this I tried to authenticate the key with Pixel 5 (Android 13) using NFC, **the phone still does not recognize the key through NFC, it does not even throw an error message**.

Maybe you can try to reproduce this for your keys.

**Comment by Jackenmen** [comment]
My USB-A Solo2 key works without issues when connected through USB-C port in my Samsung S20 phone (using a USB-C to USB-A adapter that I always carry with me) but I don't think I've ever been able to successfully use it through NFC :/ I'm not sure if having the pin on the key is the issue here since it does work through the USB port and the services it doesn't work with include the ones I set up before having a pin on the key (which means the pin isn't requested) but that's quite problematic if it is since some services force you to set one up (+ for passkeys you would need one anyway).

**Comment by SpiderD555** [comment]
Recently I have been trying to get NFC to work (on Pixel 4a) with Solokey2. I tried getting it to work on webauthn.io demo webpage (hosted by duo-labs) and to my surprise it didn't work. It did work on webauthn.me demo page (the one that is hosted by Okta). After some troubleshooting it turned out that the Solokey2 shows "bt" as transport and that makes the web browser throw an error. See the [related issue comment here](https://github.com/duo-labs/webauthn.io/issues/109#issuecomment-2023186160). I think that some small PR to the firmware code could potentially fix the issue here, but I have no knowledge about Rust (and where exactly fix it). I am leaving this comment here, so someone with more knowledge could take it from here.
